src/lib.rs

```diff
@@ -25,6 +22,9 @@ macro_rules! serde_visitor {
             fn visit_str<E>(self, s: &str) -> ::std::result::Result<Self::Value, E>
             where E: ::serde::de::Error,
             {
+                #[allow(non_upper_case_globals)]
+                const $name: &'static [&'static str] = &[$(stringify!($variant)),+];
```

Comment:
Actually I was just thinking, since this constant is local to the function, we don't need to define it as `$name` anymore, we could just use `VARIANTS` and we won't run into any collisions since the constant is local to the function now.